functional/exec_test.go

```diff
@@ -13,3 +13,104 @@
 // limitations under the License.
 
 package functional
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/clearcontainers/tests"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+)
+
+var sleepingContainerWorkload = []string{"sh", "-c", "sleep 1000"}
+
+func execDetachTiming(detach bool) TableEntry {
+	comparator := ">="
+	if detach {
+		comparator = "<"
+	}
+
+	timeout := 1
+
+	process := Process{
+		Workload: []string{"/bin/sleep", fmt.Sprintf("%d", timeout)},
+		Detach:   detach,
+	}
+
+	return Entry(fmt.Sprintf("check time as detach=%t", detach), process, comparator, timeout)
+}
+
+func execDetachOutput(detach bool) TableEntry {
+	outputHelloWorld := "HelloWorld"
+
+	process := Process{
+		Workload: []string{"echo", outputHelloWorld},
+		Detach:   detach,
+	}
+
+	expectedOutput := outputHelloWorld
+	if detach {
+		expectedOutput = ""
+	}
+
+	return Entry(fmt.Sprintf("check output as detach=%t", detach), process, expectedOutput)
+}
+
+var _ = Describe("exec", func() {
+	var (
+		container *Container
+		exitCode  int
+		err       error
+	)
+
+	BeforeEach(func() {
+		container, err = NewContainer(sleepingContainerWorkload, true)
+		Ω(container).ShouldNot(BeNil())
+		Ω(err).ShouldNot(HaveOccurred())
+		_, _, exitCode := container.Run()
+		Ω(exitCode).Should(Equal(0))
+	})
+
+	AfterEach(func() {
+		Ω(container.Exist()).Should(BeTrue())
+		_, _, exitCode = container.Delete(true)
+		Ω(exitCode).Should(Equal(0))
+		Ω(container.Cleanup()).Should(Succeed())
+	})
+
+	DescribeTable("container",
```

Comment:
It would be good to have a comment for each test block/type/grouping - such as here, it takes some effort to read the code here and the related code it is calling to try and work out it is testing that a `sleep` in a detach takes as long as is expected.
Side note - I think you are doing the sleep test for 1s, and then checking the duration below. `sleep` is not really a very accurate tool - do not be surprised if we get spurious failures on this test later with the time being <1s or >1s due to granularity and other issues. Just a heads up.